mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp

```diff
@@ -279,6 +280,31 @@ LogicalResult SYCLIDConstructorOp::verify() {
   return emitBadSignatureError(*this);
 }
 
+/// Fold:
+/// ```
+/// %c0 = arith.constant 0 : index
+/// sycl.id.constructor(%c0+) : (index+) -> memref<?x!sycl_id_X_>
+/// ```
+/// To:
+/// ```
+/// sycl.id.constructor() : () -> memref<?x!sycl_id_X_>
+/// ```
+OpFoldResult SYCLIDConstructorOp::fold(FoldAdaptor adaptor) {
+  mlir::OperandRange args = getArgs();
+  if (args.empty())
+    // Already dealing with the default constructor
+    return {};
+
+  // Check all arguments are a constant 0
+  auto isZero = m_Zero();
+  if (!llvm::all_of(args, [&](Value arg) { return matchPattern(arg, isZero); }))
```

Comment:
Can we use the capabilities of the `FoldAdaptor` here? From the documentation:
> In the above, for each method a FoldAdaptor is provided with getters for each of the operands, returning the corresponding constant attribute

Reply:
Good point. I wished [Matchers.h](https://github.com/intel/llvm/blob/f05ca70fe691bb7a231245940f0407502f2cf45d/mlir/include/mlir/IR/Matchers.h) provided something like `m_Zero` for attributes...